apps/barista-design-system/src/shared/services/universal.interceptor.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Inject, Injectable, Optional } from '@angular/core';
+import { REQUEST } from '@nguniversal/express-engine/tokens';
+import { Request } from 'express';
+import { Observable } from 'rxjs';
+
+/**
+ * In a Universal app, HTTP URLs must be absolute.
+ * This means we have to use absolute URLs when running on the server
+ * and relative URLs when running in the browser.
```

Comment:
Why do they have to be relative in the browser? They should work as absolute paths in the browser as well, right?

Reply:
universal rendering needs this http interceptor:

https://angular.io/guide/universal#using-absolute-urls-for-server-requests